codecs/ZDA.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * === GST - UTC day, month, and year, and local time zone offset ===
+ *
+ * ------------------------------------------------------------------------------
+ *        1         2  3  4    5  6   7
+ *        |         |  |  |    |  |   |
+ * $--GST,hhmmss.ss,dd,mm,yyyy,zz,zz,*hh<CR><LF>
+ * ------------------------------------------------------------------------------
+ *
+ * Field Number:
+ * 1. UTC time
+ * 2. Day, ranging between 01 and 31
+ * 3. Month, ranging between 01 and 12
+ * 4. Year
+ * 5. Local time zone offset from GMT, ranging from 00 through ±13 hours
+ * 6. Local time zone offset from GMT, ranging from 00 through 59 minutes
+ * 7. Checksum
+ */
+
+import { parseTime } from "../helpers";
+
+export const sentenceId: "ZDA" = "ZDA";
+export const sentenceName = "Data and Time";
+
+export interface ZDAPacket {
+    sentenceId: "ZDA";
+    sentenceName: string;
+    talkerId?: string;
+    datetime: Date;
+    localHoursOffset: number;
+    localMinutesOffset: number;
+}
+
+export function decodeSentence(fields: string[]): ZDAPacket {
+    return {
+        sentenceId: sentenceId,
+        sentenceName: sentenceName,
+        datetime: parseTime(fields[1], fields.slice(2, 5).join("")),
+        localHoursOffset: parseInt(fields[5], 10),
+        localMinutesOffset: parseInt(fields[6], 10)
```

Comment:
Can you use `parseIntSafe` here?  Also, should we perhaps wrap all of these up into the `Date` object with the correct timezone offset?

Reply:
Yes, you are right, I'll do that tomorrow when I'm back at my desk.

Reply:
Just a second thought @101100 about the time zone. The Date object will already know my local time zone so, correct me if I'm wrong, the time zone provided by the GPS wouldn't add any value in the Date object. That's why I just added offsets as separate props. If the GPS know its current local time zone it would probably be the same as mine. If there is a difference then my vessel/drone is located somewhere else in the world (or maybe we have a configuration problem).

Reply:
I'm OK with leaving it separate.  it just seemed that since the whole point of this sentence seems to be to get accurate time in the correct time zone we should correct in case this is running on a computer that is set to the wrong time zone.  It would be fairly easy to do that after the fact, though.  Parsing the sentence at all is better than before.

Either way, please use `parseIntSafe` as it handles blank values.

Reply:
I leave the time zone as separate props if that's ok. Can't find a way to convince Date to accept another time zone than my current one, getTimezoneOffset() will always returns my system default independent of how the Date object is initialized like...

```
new Date('2020-04-06T19:30:00+07:00').getTimezoneOffset()
  => -120

new Date('2020-04-06T19:30:00+07:00').toString() => 
  => 'Mon Apr 06 2020 14:30:00 GMT+0200 (Central European Summer Time)'
```

Reply:
Thanks for investigating!